### DISCUSSION TITLE
Ways to hide the mouse pointer using Cartesian params

### DISCUSSION THREAD

**Discussion by pevogam** [avocado-framework/avocado-vt#3655] [Q&A] [answered]
Hi all,

Is there any easy way to disable showing the mouse cursor using appropriate parameters from the Cartesian config? I have a vm like
```
[stdlog] /usr/bin/qemu-system-x86_64 \
[stdlog]     -S  \
[stdlog]     -name 'mro91' \
[stdlog]     -machine pc,memory-backend=mem-machine_mem  \
[stdlog]     -nodefaults \
[stdlog]     -device cirrus-vga,bus=pci.0,addr=0x2 \
...
[stdlog]     -cpu 'EPYC' \
...
[stdlog]     -device usb-tablet,id=usb-tablet1,bus=usb1.0,port=1 \
...
[stdlog]     -vnc :1  \
```
so I wonder if there is a VNC option (none I could find) or a change in the choice of USB tablet or perhaps change in the choice of cirrus-vga that could help me eliminate the mouse pointer from the display buffer.

@YongxueHong @luckyh Do you happen to have needed something like this in the past?

**Comment by luckyh** [comment]
Hi @pevogam , there are two mouse cursors where one is on the host and the other is on the guest, may I know which is the one you're interested in?

Assumed that you were asking about the one on the host, it should be controlled by the VNC client and showed up on the window per the tool's manner. As Qemu just held the VNC server end, I don't think there would be some room for the tweaking if I understood that correctly.

While if you meant the other, the guest os should be in charge of that, respecting some desktop settings or maybe even some device hardware settings, which was fully depend on the environment you used.

**Reply by pevogam** [reply]
Indeed, this was a good question that I should have answered to from the very beginning and I am referring to a vm setting to change the mouse cursor on the server side. We all have more control over the clients and stuff like this is easier to configure there but the server could impose its own unremovable cursor. So what I was referring to is ways to configure Avocado VT's Cartesian params in order to modify properties of the VM, either VNC or display related, in order to change such behavior. Correct me if I am wrong but it seems there are no such options right off the bat and Qemu has some command line options for specific display options.

**Comment by pevogam** [comment, marked as answer]
What could modify and in my case remove the mouse cursor was to change the VGA hardware:
```diff
                 password = test1234
-                vga = cirrus
+                vga = virtio
                 shell_prompt = "^\[.*\][\#\$]\s*$"
                 username = root
```
Not sure if VirtIO actually provides more control over this behavior but it did the trick for me, if there are any better options let me know.
